packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,196 @@
+
+type PackageJson = {
+  name: string;
+  strapi?: {
+    uuid: string;
+  };
+};
+
+async function upload(
+  ctx: CLIContext,
+  project: ProjectInfos,
+  token: string,
+  maxProjectFileSize: number
+) {
+  const cloudApi = await cloudApiFactory(token);
+  // * Upload project
+  try {
+    const storagePath = await getTmpStoragePath();
+    const projectFolder = path.resolve(process.cwd());
+    const packageJson = (await loadPkg(ctx)) as PackageJson;
+
+    if (!packageJson) {
+      ctx.logger.error(
+        'Unable to deploy the project. Please make sure the package.json file is correctly formatted.'
+      );
+      return;
+    }
+
+    ctx.logger.log('📦 Compressing project...');
+    // hash packageJson.name to avoid conflicts
+    const hashname = crypto.createHash('sha512').update(packageJson.name).digest('hex');
+    const compressedFilename = `${hashname}.tar.gz`;
+    try {
+      ctx.logger.debug(
+        'Compression parameters\n',
+        `Storage path: ${storagePath}\n`,
+        `Project folder: ${projectFolder}\n`,
+        `Compressed filename: ${compressedFilename}`
+      );
+      await compressFilesToTar(storagePath, projectFolder, compressedFilename);
+      ctx.logger.log('📦 Project compressed successfully!');
+    } catch (e: unknown) {
+      ctx.logger.error(
+        '⚠️ Project compression failed. Try again later or check for large/incompatible files.'
+      );
+      ctx.logger.debug(e);
+      process.exit(1);
+    }
+
+    const tarFilePath = path.resolve(storagePath, compressedFilename);
+    const fileStats = await fse.stat(tarFilePath);
+
+    if (fileStats.size > maxProjectFileSize) {
+      ctx.logger.log(
+        'Unable to proceed: Your project is too big to be transferred, please use a git repo instead.'
+      );
+      try {
+        await fse.remove(tarFilePath);
+      } catch (e: any) {
+        ctx.logger.log('Unable to remove file: ', tarFilePath);
+        ctx.logger.debug(e);
+      }
+      return;
+    }
+
+    ctx.logger.info('🚀 Uploading project...');
+    const progressBar = ctx.logger.progressBar(100, 'Upload Progress');
+
+    try {
+      const { data } = await cloudApi.deploy(
+        { filePath: tarFilePath, project },
+        {
+          onUploadProgress(progressEvent) {
+            const total = progressEvent.total || fileStats.size;
+            const percentage = Math.round((progressEvent.loaded * 100) / total);
+            progressBar.update(percentage);
+          },
+        }
+      );
+
+      progressBar.update(100);
+      progressBar.stop();
+      ctx.logger.success('✨ Upload finished!');
+      return data.build_id;
+    } catch (e: any) {
+      progressBar.stop();
+      if (e instanceof AxiosError && e.response?.data) {
+        if (e.response.status === 404) {
+          ctx.logger.error(
+            `The project does not exist. Remove the ${local.LOCAL_SAVE_FILENAME} file and try again.`
+          );
```

Comment:
minor thing, but could we do this for the user instead of being manual?

It would ease a bit the following scenario:
- User deletes project from cloud
- User wants to redeploy
- We ask user: `The project does not exist, do you want to remove the current ${local.LOCAL_SAVE_FILENAME} and create a new project?`

Reply:
I think it should be something removed in second iteration. As it will be possible to switch between projects.

Reply:
Also not deleting automatically would permits support to ask for the project internal name to users that face an unwanted error here.